crates/bevy_asset/src/server/info.rs

```diff
@@ -167,20 +170,25 @@ impl AssetInfos {
         loading_mode: HandleLoadingMode,
         meta_transform: Option<MetaTransform>,
     ) -> (UntypedHandle, bool) {
-        let result =
-            self.get_or_create_path_handle_internal(path, type_id, loading_mode, meta_transform);
-        unwrap_with_context(result, type_name)
+        let result = self.get_or_create_path_handle_internal(
+            path,
+            Some(type_id),
+            loading_mode,
+            meta_transform,
+        );
+        // it is ok to unwrap because TypeId was specified above
+        unwrap_with_context(result, type_name).unwrap()
     }
 
     /// Retrieves asset tracking data, or creates it if it doesn't exist.
     /// Returns true if an asset load should be kicked off
     pub fn get_or_create_path_handle_internal(
         &mut self,
         path: AssetPath<'static>,
-        type_id: TypeId,
+        type_id: Option<TypeId>,
```

Comment:
Technically this a breaking change, but I'm fine with it in a bug fix.

Actually, why is this "internal" method pub at all?

Reply:
Very good question! Sadly closing this up would be a breaking change so probably shouldn't happen in a patch release :)

Reply:
Although I guess we're already breaking it. Screw it lets close it up. Theres no way anyone is using this.